src/UPSRateRequest.php

```diff
@@ -110,7 +110,20 @@ public function getRates(ShipmentInterface $commerce_shipment, ShippingMethodInt
       }
 
       // Shop Rates.
-      $ups_rates = $request->shopRates($shipment);
+      $ship_to = $shipment->getShipTo();
+      $ship_to_address = $ship_to->getAddress();
+      $ship_to_state = $ship_to_address->getStateProvinceCode();
+
+      // Set info for Surepost.
+      if (in_array($ship_to_state, ['AA', 'AP', 'AE'])) {
+        $shipping_service = $shipment->getService();
+        $shipping_service->setCode('93');
+
+        $ups_rates = $request->getRate($shipment);
+      }
+      else {
+        $ups_rates = $request->shopRates($shipment);
+      }
```

Comment:
Just to make sure, the Surepost service is enforced for certain states?

Reply:
These are military base addresses, so for the armed forces based in Europe, Americas, and the Pacific. Surepost is like a ground delivery service. And it basically combines UPS and USPS, where the final delivery is actually made by USPS.